.github/workflows/generate-summary.yaml

```
@@ -40,8 +40,29 @@ on:
         description: 'weekly run (Optional)'
 
 jobs:
+  check-issue-name:
+    if: always()
+    runs-on: ubuntu-20.04
+    environment: production
+    defaults:
+      run:
+        working-directory: riscv-gnu-toolchain
+    steps:
+      - name: Check name
+        id: stop-issue-generation
+        run: |
+          if [ '${{ inputs.prefix }}' != '' ] && [ '${{ issue_hash_prefix }}' == 'Testsuite Status' ];
```

I think precommit just checks for contains so we should do the same here:
https://docs.github.com/en/actions/learn-github-actions/expressions#contains
Arguably we should be checking for:
`^Testsuite Status` with regex on precommit

---

updated the riscv-gnu-toolchain to gate better. Will update precommit soon. Checking for `Testsuite Status` as the `issue_hash_prefix` should be a good enough gate here

---

If a user passes in `Testsuite Status(space)` or `Testsuite Status something` we'll allow that through. I'd prefer similar to [this](https://github.com/patrick-rivos/gcc-postcommit-ci/blob/e859a7574baaffb74f4bb307af59b633f3a894d3/.github/workflows/build-with-checking.yaml#L27) on the step:
```suggestion
    if: ${{ inputs.prefix != '' && contains(inputs.issue_hash_prefix, 'Testsuite Status') }}
```

---

The point of this gate is to prevent naming schemes that would overlap with the run-frequent naming scheme. Run-Frequent passes `Testsuite Status` as the `issue_hash_prefix` and also passes `''` in as the `prefix`. Therefore, if we have `prefix != ''` and `issue_hash_prefix == 'Testsuite'status'`, this will overlap and precommit can end up using the hash. Checking `issue_hash_prefix` specifically guarantees the name of the issue https://github.com/patrick-rivos/gcc-postcommit-ci/blob/main/.github/workflows/generate-summary.yaml#L232

---

edit: https://github.com/patrick-rivos/gcc-postcommit-ci/blob/main/.github/workflows/generate-summary.yaml#L264 is where the issue name is created